freqtrade/optimize/backtesting.py

```diff
@@ -333,8 +333,12 @@ def _get_ohlcv_as_lists(self, processed: Dict[str, DataFrame]) -> Dict[str, Tupl
                     df_analyzed.loc[:, col] = 0 if not tag_col else None
 
             # Update dataprovider cache
-            self.dataprovider._set_cached_df(pair, self.timeframe, df_analyzed, CandleType.SPOT)
-            # TODO-lev: Candle-type should be conditional, either "spot" or futures
+            self.dataprovider._set_cached_df(
+                pair,
+                self.timeframe,
+                df_analyzed,
+                CandleType.FUTURES if self.trading_mode == TradingMode.FUTURES else CandleType.SPOT
```

Comment:
i'd use the variable from the config instead of this (`candle_type_def`) - which will have the correct candle type without having this logic all over the place.